src/utils/Cypher.ts

```diff
@@ -145,4 +139,110 @@ export default class Cypher {
       }
     }
   }
+
+  // This method will be reused in a Scheduler task that resumes migation
+  public static async handleCryptoSettingsChange(tenantID: string): Promise<void> {
+    const createDatabaseLock = LockingManager.createExclusiveLock(tenantID, LockEntity.DATABASE, 'migrate-settings-sensitive-data');
+    if (await LockingManager.acquire(createDatabaseLock)) {
+      try {
+        await Cypher.migrate(tenantID);
```

Comment:
The migration is synchronous right now? And will be make asynchronous after, right?

Reply:
When changing a key from the dashboard and calling the REST API yes we wait for the migration to be executed and the error is thrown and shown in dashboard if migration is not completed. Is this not what the expected behavior should be? When should it be asynchronous?

Reply:
The migration shall be run only by the crypto migration scheduler task at night, asynchronously. When a crypto settings occurs on a tenant, it just flag the boolean migrationDone to false. The REST settings endpoint will then also forbid any further crypto settings change and the dashboard behaviour will change regarding that value (grey input, etc.)

Reply:
@LucasBrazi06 I understood from our meeting and the email with the minutes that when we chance a crypto key in the dashboard we also trigger a migration of the sensitive data. Should I change to only set migrationToBeDone to true and let the scheduler task do the migration? In the current implementation both the REST endpoint and the scheduler task call the same method which executes the migration

Reply:
I'm ok to have a synchronous migration call in the REST endpoint as this is not time consuming.
If this one fails, the scheduler will take over (checks once every minute) and re-run it.
So the current implementation is ok.